### DISCUSSION TITLE
丘丘音乐的 API 跑路了

### DISCUSSION THREAD

**Discussion by Arondight** [Arondight/Adachi-BOT#434] [General]
https://github.com/Arondight/Adachi-BOT/blob/eff322f92caaec371e84716a652efdef4e485974/src/plugins/music/data.js#L21

真是个悲伤的故事

**Comment by mark9804** [comment]
我们这里有一个 API 被标记成`deprecated`了

https://github.com/Arondight/Adachi-BOT/blob/eff322f92caaec371e84716a652efdef4e485974/src/plugins/music/data.js#L68

> The `querystring` API is considered Legacy. 
> While it is still maintained, new code should use the `URLSearchParams` API instead.

**Reply by Arondight** [reply]
这类错误怎么看的，是什么代码检查工具报的么

**Reply by mark9804** [reply]
VSCode
![image](https://user-images.githubusercontent.com/9006264/143603322-fd4bac43-1c0b-4f1a-b669-76807b51e5bf.png)

**Reply by Arondight** [reply]
看看什么插件，我也整一个

**Reply by Arondight** [reply]
eslint 就可以，明天就整上一个 https://www.npmjs.com/package/eslint-plugin-deprecation
